src/Service/RequestForwarder.cpp

```diff
@@ -37,24 +38,34 @@ void RequestForwarder::run(RunnerId runner_id)
         {
             try
             {
-                if (!server->isLeader() && server->isLeaderAlive())
+                if (server->isLeader())
                 {
-                    auto client = server->getLeaderClient(runner_id);
-                    if (client)
-                    {
-                        client->send(request_for_session);
-                    }
-                    else
-                    {
-                        throw Exception("Not found client for runner " + std::to_string(runner_id), ErrorCodes::RAFT_FORWARDING_ERROR);
-                    }
+                    LOG_WARNING(log, "A leader switch may have occurred suddenly");
+                    throw Exception("Can't forward request", ErrorCodes::RAFT_ERROR);
                 }
-                else
+
+                if (!server->isLeaderAlive())
                     throw Exception("Raft no leader", ErrorCodes::RAFT_ERROR);
+
+                int32_t leader = server->getLeader();
+                ptr<ForwardingConnection> connection;
+                {
+                    std::lock_guard<std::mutex> lock(connections_mutex);
+                    connection = connections[leader][runner_id];
+                }
+
+                if (!connection)
+                    throw Exception("Not found connection for runner " + std::to_string(runner_id), ErrorCodes::RAFT_FORWARDING_ERROR);
```

Comment:
Use detailed error code?  For example here we should use RAFT_FWD_NO_CONN

Reply:
done.